DABI.js

```diff
@@ -0,0 +1,321 @@
+            }
+            if (!item || !url) continue;
+
+            items[url] = item;
+        }
+    }
+    return items;
+}
+
+function scrape(doc, url) {
+    var newItem = new Zotero.Item('journalArticle');
+    var trs = doc.getElementsByTagName("tr"),
+        headers,
+        contents,
+        items = {};
+
+    //For Loop to populate "items" Object and save tags to an Array.
+    for (var i = 0; i < trs.length; i++) {
+        headers = trs[i].getElementsByTagName("th")[0].textContent;
+        contents = trs[i].getElementsByTagName("td")[0].innerHTML;
+
+        items[headers.replace(/\s+/g, '')] = contents.trim();
+    }
+
+    //set url to fulltext resource, if present; else to database item
+    if (items["URL"] == '') {
+        newItem.url = url;
+    } else {
+        var link = doc.createElement('a');
+        link.innerHTML = items["URL"];
+        newItem.url = link.firstChild.getAttribute("href");
+
+        if (/\.pdf(#.*)?$/.test(newItem.url)) {
+            newItem.attachments = [{
+                url: newItem.url,
+                title: "DABI Full Text PDF",
+                mimeType: "application/pdf"
+            }];
+        }
+    }
+
+
+    //Formatting and saving "title" fields
+    if (items["Titel"]) {
+        newItem.title = items["Titel"].replace(/\*/g, '');
+        var short = newItem.title.replace(/^\W?(?:Die |Der |Das |\.{3}\s?)?/, '');
+        short = short.replace(/\W?(,|:|\?|!|\.|\"|\')/, '').split(' ').slice(0, 6).join(' ');
+        newItem.shortTitle = short.substring(0, 1).toUpperCase() + short.slice(1);
+        if (items["Untertitel"]) {
+            if (/(\?|!|\.)\W?$/.test(newItem.title)) {
+                newItem.title += " " + items["Untertitel"];
+            } else {
+                newItem.title += ": " + items["Untertitel"];
+            }
+        }
+    }
+
+    //Sometimes titles are missing
+    if (items["Untertitel"] && !items["Titel"]) {
+        newItem.title = items["Untertitel"].replace(/\*/g, '');
+        var short = newItem.title.replace(/^\W?(?:Die |Der |Das |\.{3}\s?)?/, '');
+        short = short.replace(/\W?(,|:|\?|!|\.|\"|\').*$/, '').split(' ').slice(0, 6).join(' ');
+        newItem.shortTitle = short.substring(0, 1).toUpperCase() + short.slice(1);
+    }
+
+    //Formatting and saving "Author" field
+    if (items["Autoren"]) {
+        var authors = items["Autoren"].split("; ");
+        for (var i = 0; i < authors.length; i++) {
+            newItem.creators.push(ZU.cleanAuthor(authors[i], "author", true));
+        }
+    }
+
+    //Formatting and saving "pages" field
+    if (items["Anfangsseite"]) {
+        newItem.pages = items["Anfangsseite"] + (items["Endseite"] ? "-" + items["Endseite"] : "");
+    }
+
+    //Saving the tags to Zotero
+    if (items["Schlagwörter"]) {
+        var tags = items["Schlagwörter"].split("; ");
+        for (var i = 0; i < tags.length; i++) {
+            newItem.tags.push(tags[i]);
+        }
```

Comment:
`newItem.tags = items["Schlagwörter"].split("; ");` is fine.

Reply:
done